artifact-management/commands/gem-test.yml

```diff
@@ -1,44 +1,87 @@
-description: Perform the actual gem build
+description: Perform gem test with default only or many Ruby versions
 parameters:
-  bundler_version:
-    description: The Bundler version that should be used for "bundle install" and the actual test command
-    type: string
-    default: 1.17.3
-  gem_repo_host:
-    description: The host to which built packages should be pushed. 
+  packagecloud_read_token:
+    description: The token to be used when fetching packages from Packagecloud.
     type: env_var_name
-    default: AVVO_GEM_REPO_HOST
+    default: PACKAGECLOUD_READ_TOKEN
   gem_repo_key:
     description: The :rubygems_api_key used to access the gem repository host
     type: env_var_name
     default: AVVO_GEM_REPO_TOKEN
+  bundler_1_version:
+    description: The Bundler version that should be used for "bundle install" and the actual test command
+    type: string
+    default: 1.17.3
+  bundler_2_version:
+    description: The Bundler version that should be used for "bundle install" and the actual test command
+    type: string
+    default: 2.1.4
   gem_test_dir:
-    description: The directory in which the gem tesets should run
+    description: The directory in which the gem tests should run
     type: string
     default: .
-  packagecloud_read_token:
-    description: The token to be used when fetching packages from Packagecloud.
-    type: env_var_name
-    default: PACKAGECLOUD_READ_TOKEN
   test_command:
-    description: The command that should be used to run tests
+     default: bundle exec rake test
+     description: The command that should be used to run tests
+     type: string
+  ruby_versions:
+    description: The ruby versions to be used to test the gem.
     type: string
-    default: bundle exec rake test
+    default: ""
 steps:
   - setup-gem-environment
-  - run:
-      name: Install bundler gem
-      command: |-
-        cd <<parameters.gem_test_dir>>
-        gem install bundler --version=<<parameters.bundler_version>>
-  - run:
-      name: bundle install
-      command: |-
-        cd <<parameters.gem_test_dir>>
-        bundle install
-  - run:
-      name: Run tests
-      command: |-
-        cd <<parameters.gem_test_dir>>
-        <<parameters.test_command>>
+  - when:
+      condition: <<parameters.ruby_versions>>
+      steps:
+        - run:
+            name: Install bundler gem
+            command: |-
+              for v in $(echo "<<parameters.ruby_versions>>" | sed "s/,/ /g");
+              do \
+                if [[ "$v" > "2.6.9999" ]];
```

Comment:
Additional tests:

```
$ if (( 17 > 3 )); then echo bigger; else echo smaller; fi
bigger
$ if (( 17.1 > 17.0 )); then echo bigger; else echo smaller; fi
bigger
$ if (( 17.1 > 17.2 )); then echo bigger; else echo smaller; fi
smaller
$ if (( "17.1" > "17.2" )); then echo bigger; else echo smaller; fi
zsh: bad math expression: illegal character: "
smaller
$ if (( 17.1 > 17.2 )); then echo bigger; else echo smaller; fi 
smaller
$ v1=17.1
$ v2=17.2
$ if (( $v1 > $v2 )); then echo bigger; else echo smaller; fi
smaller
$ v1=17.3                                                    
$ if (( $v1 > $v2 )); then echo bigger; else echo smaller; fi
bigger
```

Reply:
Also please note: **you're assuming these tests will run under bash**. Is it always the case that `gem-test` runs under bash? Be very careful about this assumption because it depends on which docker image is being used to run the tests, I think.

Reply:
From https://github.com/CircleCI-Public/slack-orb/blob/staging/src/commands/approval.yml#L48-L54 I take the idea of giving an error if there is no bash, this is because I could not found how to “require bash”. Then for both, when branch and unless branch, is specifed that /bin/bash must be used as shell.

Reply:
Yes, please use this approach.

I believe you only need to check for bash when you are using the sophisticated numeric comparison operators we are discussing elsewhere in this PR conversation, but I don't have a problem with it being a  universal requirement for all of the build upgrades we will be doing.